From dd378c0f937187c002fc65bb38ae932d985b87d2 Mon Sep 17 00:00:00 2001
From: Jenny TC <jenny.tc@intel.com>
Date: Fri, 20 Feb 2015 08:25:31 +0530
Subject: [PATCH] psy_charger: Fix online =1 on disconnect

On disconnecting charger the status is displayed as Not Charging and online
as one. This is fixed with the following chnage.
Processed the EVENT_ENUMERATE only for SDP cable. EVENT_ENUMERATE is send
for both SDP and CDP. But USB driver doesn't distinguish between SDP and CDP
and sends EVENT_ENUMERATE with 500/900mA depending on USB2/USB3. The CDP/SDP
is identified by power source detection driver which interacts with BC1.2
detection hardware. So the capabilities of CDP is decided by power source
detection driver and not by USB driver. So the EVENT_ENUMERATE needs to be
processed only for SDP.

Change-Id: I4ea65cc0690644c300317e65a12c1b2e04db62c1
Signed-off-by: Jenny TC <jenny.tc@intel.com>
---
 drivers/power/power_supply_charger.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/power/power_supply_charger.c b/drivers/power/power_supply_charger.c
index 062e1cf..c4c3cdc 100644
--- a/drivers/power/power_supply_charger.c
+++ b/drivers/power/power_supply_charger.c
@@ -167,11 +167,21 @@ static int handle_cable_notification(struct notifier_block *nb,
 				   unsigned long event, void *data)
 {
 	struct power_supply_cable_props cap;
+	struct charger_cable *cable = NULL;
 
 	if (event != USB_EVENT_CHARGER && event != PSY_CABLE_EVENT &&
 				event != USB_EVENT_ENUMERATED)
 		return NOTIFY_DONE;
 
+
+	/* Process USB_EVENT_ENUMERATED only if SDP cable is present */
+	if (event == USB_EVENT_ENUMERATED) {
+		cable = get_cable(POWER_SUPPLY_CHARGER_TYPE_USB_SDP);
+		if (cable->cable_props.cable_stat
+			== EXTCON_CHRGR_CABLE_DISCONNECTED)
+			return NOTIFY_DONE;
+	}
+
 	if (!data)
 		return NOTIFY_DONE;
 
-- 
1.9.1

